improver_tests/spotdata/test_SpotLapseRateAdjust.py

```diff
@@ -245,6 +273,112 @@ def test_xy_ordered_lapse_rate_cube(self):
         )
         self.assertArrayEqual(result.data, expected)
 
+    def test_probability_cube(self):
+        """Ensure that the plugin exits with value error if the spot data cube
+        is in probability space. """
+
+        diagnostic_cube_hash = create_coordinate_hash(self.lapse_rate_cube)
+        data = np.ones((3, 3, 3), dtype=np.float32)
+        threshold_points = np.array([276, 277, 278], dtype=np.float32)
+        probability_cube = set_up_probability_cube(
+            data, threshold_points, spp__relative_to_threshold="above"
+        )
+
+        probability_cube.attributes["model_grid_hash"] = diagnostic_cube_hash
+
+        plugin = SpotLapseRateAdjust()
+        msg = (
+            "Input cube has a probability coordinate which cannot be lapse "
+            "rate adjusted. Input data should be in percentile or "
+            "deterministic space only."
+        )
+
+        with self.assertRaisesRegex(ValueError, msg):
+            plugin(probability_cube, self.neighbour_cube, self.lapse_rate_cube)
+
+    def test_different_dimensions(self):
+        """Test that the lapse rate cube can be broadcast to the same dimensions
+        as the spot data cube."""
+
+        data = np.array([25, 50, 75], dtype=np.float32)
+        spot_temperature_new_coord = add_coordinate(
+            self.spot_temperature_nearest, data, "percentile", "%"
+        )
+        plugin = SpotLapseRateAdjust()
+        result = plugin(
+            spot_temperature_new_coord, self.neighbour_cube, self.lapse_rate_cube
+        )
+        expected = np.array([280 + (2 * DALR), 270, 280 - DALR]).astype(np.float32)
+
+        self.assertArrayEqual(result[0].data, expected)
```

Comment:
Why haven't you checked that you still have a percentile coord and that the result on each percentile is as expected?

Reply:
Done.